Cards/CardBase.gd

```diff
@@ -35,12 +35,41 @@ func _apply_all_effects(target: Entity) -> void:
 	for effect_data: EffectData in card_effects_data:
 		effect_data.apply_effect_data(target)
 
+func _apply_effect(target : Entity, effect: EffectData):
+	effect.apply_effect_data(target)
 
 func can_play_card(caster: Entity, target: Entity) -> bool:
 	return caster.get_party_component().can_play_on_entity(application_type, target)
 
 
-func on_card_play(caster: Entity, target: Entity) -> void:
-	_apply_all_effects(target)
+func on_card_play(caster: Entity, targets: Array[Entity]) -> void:
```

Comment:
this honestly feels like it could be moved in the `_apply_all_effects` function, because this is basically what it does

Reply:
After reading again, it does more than just `_apply_all_effects, so it's fine